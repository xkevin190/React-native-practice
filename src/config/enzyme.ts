import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

/**
 * @tech-debt
 *
 * Update enzyme to support react 18
 */
Enzyme.configure({ adapter: new Adapter() });
