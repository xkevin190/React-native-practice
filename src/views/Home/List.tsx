import React, {useCallback, useState} from 'react';
import {View, StyleSheet, FlatList} from 'react-native';
import Item from '../../components/Item';
import {NavigationProps} from '../../constants/types';

import {item as IItem, ListItems} from '../../state/aplication/type';

interface ListProps {
  ListItems?: ListItems;
  route?: any;
  loadingApp?: boolean;
  setFavorite: (name: string) => void;
}

type renderItemProps = {
  item: IItem;
  index: number;
};

const keyExtractor = (_item: IItem) => _item.name;

const List = (props: ListProps) => {
  const action = (url: string) => {
    props.setFavorite(url);
  };

  const memoizedCallback = useCallback(action, []);

  const renderItem = ({item}: renderItemProps): JSX.Element => {
    return (
      <Item
        gender={item.gender}
        name={item.name}
        homeworld={item.homeworld}
        birth_year={item.birth_year}
        favorite={item.favorite}
        action={memoizedCallback}
      />
    );
  };

  return (
    <FlatList
      data={props.ListItems}
      keyExtractor={keyExtractor}
      renderItem={renderItem}
      removeClippedSubviews={true}
      initialNumToRender={4}
    />
  );
};

export default List;
