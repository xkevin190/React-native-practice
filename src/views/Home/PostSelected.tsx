import * as React from 'react';
import {View, StyleSheet} from 'react-native';
import Header from '../../components/Header';
import {NavigationProps} from '../../constants/types';
import Webview from 'react-native-webview';
import {RouteProp} from '@react-navigation/native';

interface PostSelectedProps {
  navigation: NavigationProps;
  route: RouteProp<any, any>;
}

const PostSelected = (props: PostSelectedProps) => {
  return (
    <View style={styles.container}>
      <Header back={true} navigation={props.navigation} />
      <Webview
        source={{uri: `https://www.reddit.com${props.route.params.url!}`}}
      />
    </View>
  );
};

export default PostSelected;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
