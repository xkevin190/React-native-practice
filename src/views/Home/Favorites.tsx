import React, {useState} from 'react';
import {connect} from 'react-redux';
import {IState} from '../../state/root';
import Container from '../../components/Container';
import Header from '../../components/Header';
import {View} from 'react-native';
import {ActionCreator} from 'redux';
import {getImages, setFavorite} from '../../state/aplication/action';
import {NavigationProps} from '../../constants/types';
import {ListItems} from '../../state/aplication/type';
import List from './List';
import Input from '../../components/TextInput';

interface FavoritesProps {
  navigation?: NavigationProps;
  listItems: ListItems;
  _setFavorite: (name: string) => void;
}

const Favorites = ({navigation, listItems, _setFavorite}: FavoritesProps) => {
  const [search, setSearch] = useState('');
  const favorites = listItems.filter(item => {
    return (
      item.name.toLowerCase().includes(search.toLowerCase()) && item.favorite
    );
  });

  return (
    <Container>
      <View>
        <Header navigation={navigation} title={'Favorites'} />
        <Input onchange={text => setSearch(text)} />
        <List
          setFavorite={_setFavorite}
          ListItems={favorites}
          navigation={navigation!}
        />
      </View>
    </Container>
  );
};
const mapStateToProps = (state: IState) => ({
  listItems: state.aplication.items,
  loadingApp: state.aplication.loading,
});

const mapDispatchToProps = (dispatch: ActionCreator<any>) => ({
  getItems: (type: number) => dispatch(getImages(type)),
  _setFavorite: (name: string) => dispatch(setFavorite(name)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
