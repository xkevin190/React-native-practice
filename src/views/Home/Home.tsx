import * as React from 'react';
import {Text, View, StyleSheet, FlatList} from 'react-native';
import {connect} from 'react-redux';
import {NavigationProps} from '../../constants/types';
import {IAplicationState, item} from '../../state/aplication/type';
import {IState} from '../../state/root';
import Container from '../../components/Container';
import Header from '../../components/Header';
import {ActionCreator} from 'redux';
import {getImages} from '../../state/aplication/action';

import Item from '../../components/Item';
import {createMaterialTopTabNavigator} from '@react-navigation/material-top-tabs';
import List from './List';

interface HomeProps {
  navigation: NavigationProps;
  aplication: IAplicationState;
  getItems: () => void;
}

const Home = (props: HomeProps) => {
  const Tab = createMaterialTopTabNavigator();
  React.useEffect(() => {
    props.getItems();
  }, []);

  return (
    <Container>
      <>
        <Header />
        <Tab.Navigator>
          <Tab.Screen name="New" component={List} />
          <Tab.Screen name="Top" component={List} />
          <Tab.Screen name="Popular" component={List} />
          <Tab.Screen name="Hot" component={List} />
        </Tab.Navigator>
      </>
    </Container>
  );
};

const mapStateToProps = (state: IState) => ({
  aplication: state.aplication,
});

const mapDispatchToProps = (dispatch: ActionCreator<any>) => ({
  getItems: () => dispatch(getImages()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
