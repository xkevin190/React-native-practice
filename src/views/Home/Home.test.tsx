import * as React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import Home from './Home';
import enableHooks from 'jest-react-hooks-shallow';
import {View} from 'react-native';

enableHooks(jest, {dontMockByDefault: true});
describe('Home (Unit-Test)', () => {
  let wrapper: ShallowWrapper;

  //   const favoriteAction = jest.fn();
  const list = {
    aplication: {
      items: [
        {
          gender: 'test',
          name: 'test',
          birth_year: 'test',
          homeworld: 'test',
          favorite: false,
        },
      ],
    },
  };

  const store = {
    dispatch: jest.fn(),
    getState: jest.fn().mockReturnValue(list),
  };

  beforeEach(() => {
    wrapper = shallow(<Home store={store} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
