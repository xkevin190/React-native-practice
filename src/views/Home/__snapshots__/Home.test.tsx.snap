// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home (Unit-Test) should match snapshot 1`] = `
<ContextProvider
  value={null}
>
  <Home
    _setFavorite={[Function]}
    getItems={[Function]}
    listItems={
      Array [
        Object {
          "birth_year": "test",
          "favorite": false,
          "gender": "test",
          "homeworld": "test",
          "name": "test",
        },
      ]
    }
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Object {
                "aplication": Object {
                  "items": Array [
                    Object {
                      "birth_year": "test",
                      "favorite": false,
                      "gender": "test",
                      "homeworld": "test",
                      "name": "test",
                    },
                  ],
                },
              },
            },
          ],
        },
      }
    }
  />
</ContextProvider>
`;
